src/syntax.mli

```diff
@@ -345,8 +348,12 @@ val mk_false : 'a context -> 'a formula
    symbols or quantifiers. *)
 val eliminate_ite : 'a context -> 'a formula -> 'a formula
 
-(** Print a formula as a satisfiability query in SMTLIB2 format.  The query
-    includes function declarations and (check-sat). *)
+(** Print a formula as a satisfiability query in SMTLIB2 format.
+    The query includes function declarations and (check-sat). *)
+val pp_smtlib2_gen : ?named:bool -> ?env:(string Env.t) -> ?strings:((string, symbol) Hashtbl.t) ->
```

Comment:
What are the "named' and "strings" parameters?

Reply:
Named is whether or not each formula output should be named. E.g. the first assertion is named f0, then f1, ..., the nth one is named fn. If the formulas/assertions are named then you can use Smtlib2 commands to retrieve which assertions form the UNSAT core.

strings will store the mapping from Smtlib2 symbols/names to SRK symbols. This helps facilitate writing a Smtlib2 file, calling a SMT solver then retrieving the model and parsing it to a SRK interpretation.